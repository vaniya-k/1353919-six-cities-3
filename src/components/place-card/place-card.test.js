import React from 'react';
import renderer from 'react-test-renderer';
import PlaceCard from './place-card.jsx';
import {createStore} from "redux";
import {Provider} from "react-redux";
import {Router} from 'react-router-dom';
import history from '../../history.js';

const mockReducer = () => {
  const state = {
    offers: null,
    user: {
      authorizationStatus: `AUTH`
    }
  };
  return state;
};

const mockStore = createStore(mockReducer);

const place = {
  id: 4,
  title: `Wayne Estate`,
  price: 789,
  type: `apartment`,
  rating: 100,
  isPremium: true,
  isFavorite: true,
  previewUrl: `https://`,
  gps: {lat: 123, lon: 123}
};

it(`<PlaceCard/> should show Wayne Estate rendered for <FavoritesPage/>`, () => {
  const mock = jest.fn();

  const tree = renderer
    .create(<Provider store={mockStore}><Router history={history}><PlaceCard placeLatLon={{lat: 123, lon: 123}} place={place} handleHover={mock} page={`favorites`}/></Router></Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`<PlaceCard/> should show Wayne Estate rendered for <MainPage/>`, () => {
  const mock = jest.fn();

  const tree = renderer
    .create(<Provider store={mockStore}><Router history={history}><PlaceCard placeLatLon={{lat: 123, lon: 123}} place={place} handleHover={mock}/></Router></Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`<PlaceCard/> should show Wayne Estate rendered for <PlacesListNearby/> on <PlacePage>`, () => {
  const mock = jest.fn();

  const tree = renderer
    .create(<Provider store={mockStore}><Router history={history}><PlaceCard placeLatLon={{lat: 123, lon: 123}} place={place} handleHover={mock} page={`place`}/></Router></Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
