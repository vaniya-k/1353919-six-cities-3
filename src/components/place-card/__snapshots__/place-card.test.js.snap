// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlaceCardMain/> should render "Tiny apartment" 1`] = `
<article
  className="cities__place-card place-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="place-card__mark"
  >
    <span>
      Premium
    </span>
  </div>
  <div
    className="cities__image-wrapper place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src="img/apartment-03.jpg"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          70
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button place-card__bookmark-button--active button"
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            href="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          In bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "80%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="/dev-place-page"
        onClick={[MockFunction]}
      >
        Tiny apartment
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      Apartment
    </p>
  </div>
</article>
`;
